### PR TITLE
Hotfix for p7zip #2368.

### PR DESCRIPTION
**Pull request by sirjaymz**
_Motivation:  Latest version of P7zip 15.14.1 with 2 vulnerability patched from 16.02 7-zip src.
_Linked issues: https://github.com/SynoCommunity/spksrc/issues/2368

### CODE REVIEW

File: cross/p7zip/digests [hunk 1]
```diff
@@ -1,3 +1,2 @@
-p7zip_9.20.1_src_all.tar.bz2 SHA1 1cd567e043ee054bf08244ce15f32cb3258306b7
-p7zip_9.20.1_src_all.tar.bz2 SHA256 49557e7ffca08100f9fc687f4dfc5aea703ca207640c76d9dee7b66f03cb4782
-p7zip_9.20.1_src_all.tar.bz2 MD5 bd6caaea567dc0d995c990c5cc883c89
+p7zip_15.14.1_src_all.tar.bz2 SHA1 9b15a79f94230fab9b9d4f9f532c723117145c7a
+p7zip_15.14.1_src_all.tar.bz2 MD5 92cca093312b5a71a7be7dc7d1d32509
```
Comment (sirjaymz) [inline cross/p7zip/digests]: sha256 checksum is not available on Sourceforge...
So whats the procedure? Am i suppose to generate a sha256? i only provided the 2 checksums that were available by the sourceforge developers, as i would think they 'own' that deliverable of creating the checksums. thoughts?

Comment (GuillaumeSmaha) [inline cross/p7zip/digests]: Yes, you just generate the sha256 for the file.

**Comment by sirjaymz** [comment]
added sha256 checksum to the digest

File: cross/p7zip/digests (hunk 1 above)
Comment (sirjaymz) [inline cross/p7zip/digests]: added sha256 checksum to the digest

**Comment by Dr-Bean** [comment]
@sirjaymz Thanks. The individual patch contents look okay to me, but I will probably cherry-pick the correct commits to clean up the PR.

For future reference, please don't remove the checkboxes in the PR template: they're there to make sure we know if you've tested the PR and in what sense. I'll assume you didn't test.

File: cross/p7zip/digests (hunk 1 above)
Comment (Dr-Bean) [inline cross/p7zip/digests]: Use `make digests` to create a digests file (and then verify the outcome against upstream). Of course, that only works if you have a dev environment set up ;)

**Comment by Dr-Bean** [comment]
Redid the update/patches here: https://github.com/SynoCommunity/spksrc/commit/0e8a35efdbf5c07deb26fd94198fbbeb983d3cf9.
Thanks for taking the time to create a PR :+1:

**Pull request closed without merging**